src/saveload/saveload.cpp

```diff
@@ -2184,14 +2185,18 @@ struct FileReader : LoadFilter {
 	{
 		if (this->file != nullptr) fclose(this->file);
 		this->file = nullptr;
+
+		_game_session_stats.savegame_size = this->read_bytes;
 	}
 
 	size_t Read(byte *buf, size_t size) override
 	{
 		/* We're in the process of shutting down, i.e. in "failure" mode. */
 		if (this->file == nullptr) return 0;
 
-		return fread(buf, 1, size, this->file);
+		size_t res = fread(buf, 1, size, this->file);
+		read_bytes += res;
```

Comment:
During reading the file we seek through it (see `Reset()`), so you might be counting bytes double. I am not sure whether the amount of bytes counted double is significant enough to care though.
Also, we have a `begin`, so maybe we can just `_game_session_stats.savegame_size = ftell(this->file) - this->begin` with the proper guards for `file` being `nullptr`.

Reply:
Good catch! And `ftell` for both looks nicer anyway :D